### PR TITLE
Use prefix from  Eventing CR

### PR DESCRIPTION
**Pull request by muralov**
**Description**

Changes proposed in this pull request:

- Add eventTypePrefix under BackendConfig
- Add validation for eventTypePrefix
- Add validation unit tests

**Related issue(s)**
#25

### CODE REVIEW

File: pkg/env/backend_config.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	PortNum           int    `envconfig:"PUBLISHER_PORT_NUM" default:"8080"`
+	MetricsPortNum    int    `envconfig:"PUBLISHER_METRICS_PORT_NUM" default:"8080"`
+	ServiceAccount    string `envconfig:"PUBLISHER_SERVICE_ACCOUNT" default:"eventing-publisher-proxy"`
+	Replicas          int32  `envconfig:"PUBLISHER_REPLICAS" default:"1"`
+	RequestsCPU       string `envconfig:"PUBLISHER_REQUESTS_CPU" default:"32m"`
+	RequestsMemory    string `envconfig:"PUBLISHER_REQUESTS_MEMORY" default:"64Mi"`
+	RequestTimeout    string `envconfig:"PUBLISHER_REQUEST_TIMEOUT" default:"5s"`
+	LimitsCPU         string `envconfig:"PUBLISHER_LIMITS_CPU" default:"100m"`
+	LimitsMemory      string `envconfig:"PUBLISHER_LIMITS_MEMORY" default:"128Mi"`
+	PriorityClassName string `envconfig:"PUBLISHER_PRIORITY_CLASS_NAME" default:""`
```
Comment (mfaizanse) [inline pkg/env/backend_config.go:42]: Do we still need these EPP envs now?

File: pkg/env/backend_config.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	// publisher takes the controller values
+	AppLogFormat string `envconfig:"APP_LOG_FORMAT" default:"json"`
+	AppLogLevel  string `envconfig:"APP_LOG_LEVEL" default:"info"`
```
Comment (mfaizanse) [inline pkg/env/backend_config.go:45]: Do we still need these EPP envs now? or the log level comes from eventing cr?

File: pkg/env/config.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+const (
+	backendKey = "BACKEND"
+
+	backendValueBEB  = "BEB"
+	backendValueNats = "NATS"
+)
+
+// Backend returns the selected backend based on the environment variable
+// "BACKEND". "NATS" is the default value in case of an empty variable.
```
Comment (mfaizanse) [inline pkg/env/config.go:21]: Do we still need this? or it comes from eventing cr?

File: pkg/env/config.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	}
+}
+
+// Config represents the environment config for the Eventing Controller.
+type Config struct {
+	// Following details are for eventing-controller to communicate to BEB
+	BEBAPIURL     string `envconfig:"BEB_API_URL" default:"https://enterprise-messaging-pubsub.cfapps.sap.hana.ondemand.com/sap/ems/v1"`
+	ClientID      string `envconfig:"CLIENT_ID" default:"client-id"`
+	ClientSecret  string `envconfig:"CLIENT_SECRET" default:"client-secret"`
+	TokenEndpoint string `envconfig:"TOKEN_ENDPOINT" default:"token-endpoint"`
```
Comment (mfaizanse) [inline pkg/env/config.go:41]: We dont need these webhook configs after Eventing auth manager.

Comment (mfaizanse) [reply pkg/env/config.go:41]: Can we only copy fields which we need?

File: pkg/env/backend_config.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+	"github.com/kelseyhightower/envconfig"
+)
+
+// BackendConfig represents the environment config for the Backend Controller.
+type BackendConfig struct {
+	PublisherConfig PublisherConfig
+
+	BackendCRNamespace string `envconfig:"BACKEND_CR_NAMESPACE" default:"kyma-system"`
+	BackendCRName      string `envconfig:"BACKEND_CR_NAME" default:"eventing-backend"`
```
[the remark this answers is not in the document]
Comment (muralov) [reply pkg/env/backend_config.go:15]: I had it imported, but then decided to copy as the EVENT_TYPE_PREFIX, NATS_URL, etc. were required which we actually get from Eventing CR. I didn't want to make them optional in the EC.

File: pkg/env/backend_config.go (hunk 1 above)
Comment (muralov) [reply pkg/env/backend_config.go:42]: I knew you would ask this. Will remove unnecessary ones 😄

File: pkg/env/backend_config.go (hunk 2 above)
Comment (muralov) [reply pkg/env/backend_config.go:45]: I'll set the log level from Eventing it was not done yet. Let's leave the APP_LOG_FORMAT so it sets default value. What do you think?

File: test/utils/options.go [hunk 6]
```diff
@@ -100,3 +79,17 @@ func WithEventingInvalidBackend() EventingOption {
...
 }
+
+func WithEventingEventTypePrefix(eventTypePrefix string) EventingOption {
+	return func(nats *v1alpha1.Eventing) error {
+		nats.Spec.Backends[0].Config.EventTypePrefix = eventTypePrefix
+		return nil
+	}
+}
+
+func WithEventingEventingLogLevel(logLevel string) EventingOption {
```
Comment (mfaizanse) [inline test/utils/options.go:90]: Is the multiple `Eventing` word intentional?

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**